gas_station/src/lib.rs

```diff
@@ -356,8 +367,13 @@ impl Contract {
             .0;
 
         let signature_requests = vec![
-            SignatureRequest::new(&paymaster.key_path, paymaster_transaction, true),
-            SignatureRequest::new(&sender, transaction_request.clone(), false),
+            SignatureRequest::new(
+                &paymaster.key_path,
+                key_version,
+                paymaster_transaction,
+                true,
+            ),
+            SignatureRequest::new(&sender, key_version, transaction_request.clone(), false),
```

Comment:
Seems like there should actually be two key_versions passed to these two different transactions (i.e. `{sender,paymaster}_key_version`